spec/lib/karafka_spec.rb

```diff
@@ -3,82 +3,15 @@
 RSpec.describe Karafka do
   subject { described_class }
 
-  let(:std_logger) { subject.logger }
-
-  before do
-    std_logger
-  end
-
-  after do
-    subject.logger = std_logger
-  end
-
-  describe '#logger=' do
-    let(:logger) { double }
-
-    it 'assigns logger' do
-      subject.logger = logger
-      expect(subject.instance_variable_get(:'@logger')).to eq logger
+  describe '.logger' do
+    it 'expect to use app logger' do
+      expect(subject.logger).to eq described_class::App.config.logger
```

Comment:
Not really - If by some reason it would start to be an instance (not a class), we would still refer internals based on the class.